### ISSUE TITLE
Plugins: Conflict between external and core plugin

### ISSUE THREAD

**Issue by andresmgot** [grafana/grafana#43401]
**What happened**:

We found an interesting issue regarding plugins. There is this plugin https://grafana.com/grafana/plugins/grafana-azure-monitor-datasource/?tab=installation which is an old version of the Azure plugin bundled with Grafana as a core plugin (both have the same ID). If you install it, in Grafana v8.2, that plugin is ignored and the core one is used but in v8.3, the "external" plugin takes precedence.

Note that the core plugin version is 0.3.0 while the external one is at 0.3.1. I tried bumping the version of the core plugin (to 1.0.0) but the problem still persists.

**What you expected to happen**:

The core plugin to takes precedence as well in Grafana 8.3.

**How to reproduce it (as minimally and precisely as possible)**:

- Install the external plugin ([zip file](https://grafana.com/api/plugins/grafana-azure-monitor-datasource/versions/0.3.1/download)).
- Go to the plugin page (e.g. http://localhost:3000/admin/plugins/grafana-azure-monitor-datasource). It should not show the one with the version 0.3.1.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.3.3
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by marefr** [comment]
Introduced by #40477 I believe. Quite a big refactoring so this might have to go into v8.4.0. We'll pick it up again in January. There's a workaround and that is to uninstall the external plugin.

**Comment by wbrowne** [comment]
This should be solved with the following ongoing work here https://github.com/grafana/grafana/pull/43171

**Comment by marefr** [comment]
Fixed by #43171

**Comment by fridgepoet** [comment]
Hi, seems like the external plugin is still able to be installed.
Issue #46691 raised the issue to us, and I was able to reproduce with the following:
(Sorry the steps include restarting at least twice. I'm not sure why yet. This is at least what I observed and was able to reproduce consistently on my machine)

Start with no external Azure Monitor plugin installed. It should default to core version (0.3.0)

1. check out v8.4.3
1. start server (make run)
2. check Azure plugin version. should be Core version 0.3.0
3. use grafana cli: ./grafana-cli plugins install grafana-azure-monitor-datasource . observe that it downloads 0.3.1 zip
4. restart server (re-run make run)
5. check Azure plugin version. For me it shows Core 0.3.0 again
6. restart server (re-run make run)
7. check Azure plugin version. For me it shows External version 0.3.1. Any subsequent restarts continue to show 0.3.1

Feel free to close this again if I am mistaken!

Not a high priority issue, by the way, because our planned workaround is to upgrade the Core version to 1.0.0 as I have not been able to override and install the External version when the Core version was 1.0.0.
